src/app/app.module.ts

```diff
@@ -1,30 +1,94 @@
+  MdIconModule,
+  MdIconRegistry,
+  MdInputModule,
+  MdListModule,
+  MdOptionModule,
+  MdProgressSpinnerModule,
+  MdCardModule
+} from '@angular/material';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 import { StoreModule } from '@ngrx/store';
+import { EffectsModule } from '@ngrx/effects';
+import { StoreRouterConnectingModule } from '@ngrx/router-store';
 
-import { routes } from './app.routes';
 import { AppComponent } from './app.component';
 import { HeaderComponent } from './components/header/header.component';
 import { SidenavComponent } from './components/sidenav/sidenav.component';
-import { recipesReducer, loadedRecipesReducer } from './store/recipes.store';
+import { RecipesAndFiltersComponent } from './components/recipes-and-filters/recipes-and-filters.component';
+import { RecipesComponent } from './components/recipes/recipes.component';
+import { RecipeDetailsComponent } from './components/recipe-details/recipe-details.component';
+import { RecipeComponent } from './components/recipe/recipe.component';
+import { FiltersComponent } from './components/filters/filters.component';
+import { CardComponent } from './components/card/card.component';
 import { FeaturesComponent } from './components/features/features.component';
 
-let rootReducer = { recipes: recipesReducer, loadedRecipes: loadedRecipesReducer, };
+import { Backend } from './services/backend';
+import { MaterialIconsService } from './services/material-icons/material-icons.service';
+import { Recipe} from './models/recipe.model';
+import { appReducer } from './store/reducers/reducers';
+import { RecipesEffects } from './store/effects/effects';
+import { initialState } from './models/state.model';
 
 @NgModule({
   declarations: [
     AppComponent,
+    CardComponent,
+    FeaturesComponent,
     HeaderComponent,
     SidenavComponent,
-    FeaturesComponent,
+    RecipesAndFiltersComponent,
+    RecipesComponent,
+    RecipeDetailsComponent,
+    RecipeComponent,
+    FiltersComponent
   ],
   imports: [
     BrowserModule,
-    RouterModule.forRoot(routes),
-    SharedModule.forRoot(),
-    StoreModule.provideStore(rootReducer),
+    ReactiveFormsModule,
+    HttpModule,
+    NoopAnimationsModule,
+    MdInputModule,
+    MdCheckboxModule,
+    MdToolbarModule,
+    MdButtonModule,
+    MdSidenavModule,
+    MdIconModule,
+    MdListModule,
+    MdOptionModule,
+    MdSelectModule,
+    MdProgressSpinnerModule,
+    MdCardModule,
+    RouterModule.forRoot([
+      { path: '',  pathMatch: 'full', redirectTo: 'recipes' },
+      { path: 'features', component: FeaturesComponent },
+      { path: 'recipes',  pathMatch: 'full', component: RecipesAndFiltersComponent },
+      { path: 'recipe/:id', component: RecipeDetailsComponent }
+    ]),
+
+    StoreModule.forRoot(<any>{app: appReducer}, {initialState}),
+
+    EffectsModule.forRoot([
+      RecipesEffects
+    ]),
+
+    StoreRouterConnectingModule
+  ],
+  providers: [
```

Comment:
if we want to introduce lazy loading later on we need these services and some components above in a shared module. We can also do that when the time comes, we should at least have some page thats loaded lazily.